sklearn/decomposition/dict_learning.py

```diff
@@ -963,6 +982,12 @@ class SparseCoder(BaseEstimator, SparseCodingMixin):
 
         .. versionadded:: 0.20
 
+    transform_max_iter : int, optional (default=1000)
+        If `algorithm='lasso_lars'` or `algorithm='lasso_cd'`,
+        `transform_max_iter` is passed to the underlying transformer.
```

Comment:
Using the term "transformer" in this context may be slightly confusing. The sparse encoding is based off of `LassoLars`'s coefficients. Not a typical "scikit-learn transformer".

An alternative would be to state what it does: Maximum number of iterations to perform if `algorithm='lasso_cd'` or `lasso_cd`.